DisguiseUnityRenderStream/Runtime/PipelineAbstraction/SRGBConversions.cs

```diff
@@ -0,0 +1,202 @@
+            {
+                if (colorSpace == Space.Linear && textureFormat == Texture.sRGB)
+                {
+                    throw new ArgumentException($"{nameof(SRGBConversions)}: The combination of sRGB texture format" +
+                                                $"and linear color space is not supported");
+                }
+                
+                this.colorSpace = colorSpace;
+                this.textureFormat = textureFormat;
+            }
+        }
+
+        public static Texture GetTextureFormat(RenderTexture rt)
+        {
+            return GetTextureFormat(rt.graphicsFormat);
+        }
+
+        public static Texture GetTextureFormat(GraphicsFormat format)
+        {
+            return GraphicsFormatUtility.IsSRGBFormat(format)
+                ? Texture.sRGB
+                : Texture.Other;
+        }
+        
+        public static Texture GetTextureFormat(Display display)
+        {
+            return display.requiresSrgbBlitToBackbuffer
+                ? Texture.Other
+                : Texture.sRGB;
+        }
+
+        /// <summary>
+        /// Returns the descriptor for a RenderTexture generated by Unity during frame rendering.
+        /// An sRGB texture is assumed to contain <see cref="Space.sRGB"/> data.
+        /// Any other texture is assumed to contain <see cref="Space.Linear"/> data.
+        /// </summary>
+        public static Descriptor GetAutoDescriptor(RenderTexture rt)
+        {
+            return GetAutoDescriptor(rt.graphicsFormat);
+        }
+        
+        /// <inheritdoc cref="GetAutoDescriptor(RenderTexture)"/>
+        public static Descriptor GetAutoDescriptor(GraphicsFormat format)
+        {
+            var textureFormat = GetTextureFormat(format);
+            
+            return textureFormat switch
+            {
+                Texture.sRGB => new Descriptor(Space.sRGB, textureFormat),
+                Texture.Other => new Descriptor(Space.Linear, textureFormat),
+                _ => throw new ArgumentOutOfRangeException()
+            };
+        }
+        
+        public static Descriptor GetDisplayDescriptor(Display display)
+        {
+            var textureFormat = GetTextureFormat(display);
+            return new Descriptor(Space.sRGB, textureFormat);
+        }
+        
+        struct TableEntryHelper
+        {
+            public Space srcColor;
+            public Texture srcTex;
+            public Space dstColor;
+            public Texture dstTex;
+        }
+
+        /// <returns>
+        /// The software conversion to apply inside a blit shader to ensure color consistency between <paramref name="src"/> and <paramref name="dest"/>.
+        /// </returns>
+        /// <exception cref="InvalidOperationException">Invalid descriptor.</exception>
+        public static Conversion GetConversion(Descriptor src, Descriptor dst)
+        {
+            var x = new TableEntryHelper()
```

Comment:
Minor. This can be a named tuple (instead of a struct)
```
(Space srcColor, Texture srcTex, Space dstColor, Texture dstTex) x = (src.colorSpace, src.textureFormat, dst.colorSpace, dst.textureFormat);
```